command/agent/config.go

```diff
@@ -819,6 +855,15 @@ func MergeConfig(a, b *Config) *Config {
 	if b.AdvertiseAddrWan != "" {
 		result.AdvertiseAddrWan = b.AdvertiseAddrWan
 	}
+	if b.AdvertiseAddrs.SerfLan != nil {
+		result.AdvertiseAddrs.SerfLan = b.AdvertiseAddrs.SerfLan
+	}
+	if b.AdvertiseAddrs.SerfWan != nil {
+		result.AdvertiseAddrs.SerfWan = b.AdvertiseAddrs.SerfWan
+	}
+	if b.AdvertiseAddrs.RPC != nil {
+		result.AdvertiseAddrs.RPC = b.AdvertiseAddrs.RPC
+	}
```

Comment:
Looks like we are missing tests for this. Let's definitely make sure this is tested.